src/main/comments.js

```diff
@@ -306,6 +306,50 @@ function attach(comments, ast, text, options) {
   }
 }
 
+const isAllEmptyAndNoLineBreak = (text) =>
+  !/[\n\u2028\u2029]/.test(text) && /^\s*$/.test(text);
+function isOwnLineComment(comment, text, options, comments, commentIndex) {
+  const { locStart, locEnd } = options;
+  let start = locStart(comment);
+
+  // Find first comment on the same line
+  for (let index = commentIndex - 1; index >= 0; index--) {
+    const previousComment = comments[index];
+    if (!previousComment) {
```

Comment:
Let's add a check here that `comment` and `previousComment` have the same `precedingNode`. Otherwise, `slice` might return huge strings when the comments are far from each other.